### ISSUE TITLE
Fix tests after modules repository restructuring

### ISSUE THREAD

**Issue by JoseEspinosa** [nf-core/modules#2154]
After https://github.com/nf-core/modules/pull/2141 was merged tests failed in the following modules as nicely reported by @grst in [modules slack](https://nfcore.slack.com/archives/CJRH30T6V/p1665044386361599?thread_ts=1664961961.493299&cid=CJRH30T6V)
Please assign yourself by editing this issue e.g. - [ ] module/foo (@JoseEspinosa) and mark it as resolved when your PR fixing the test is merged.

- [x] amps #3244
- [x] amrfinderplus/run @rpetit3
- [x] amrfinderplus/update @rpetit3
- [x] antismash/antismashlitel (not failing, cf #2687 and #2688 )
- [x] antismash/antismashlitedownloaddatabasesl (not failing, cf #2687 and #2688 )
- [x] arriba (@JoseEspinosa)
- [x] bamcmp #3245
- [X] bases2fastq (@Emiller88 https://github.com/nf-core/modules/pull/2074)
- [x] bcftools/reheader #2497 
- [x] bcftools/sort #2497 
- [x] biscuit/epiread #2475 
- [x] checkm/qa (not failing, cf #2464)
- [x] cnvkit/batch #2492 
- [x] controlfreec/assesssignificance #3248 
- [x] controlfreec/freec #2980
- [x] controlfreec/freec2bed #3248
- [x] controlfreec/freec2circos #3248 
- [x] controlfreec/makegraph #3248 
- [x] cooler/cload @pawelciurkaardigen (https://github.com/nf-core/modules/pull/2219)
- [x] cooler/zoomify #1937 
- [x] custom/sratoolsncbisettings #3964 
- [ ] deeparg/downloaddata tracked in #4321
- [ ] deeparg/predict tracked in #4321
- [x] deepvariant #2494
- [x] delly/call #2909 and #3246
- [x] endorspy #3134 
- [x] entrezdirect/esearch #3243
- [x] ffq #3978 
- [x] flye #3977 
- [x] freebayes #2493 #3242
- [x] gatk4/applybqsr @pawelciurkaardigen (https://github.com/nf-core/modules/pull/2219)
- [x] gatk4/genotypegvcfs @pawelciurkaardigen 
- [x] gatk4/mutect2 @pawelciurkaardigen 
- [x] gtdbtk/classifywf #16665
- [x] kallistobustools/ref (not failing)
- [x] leehom @chriswyatt1 (need PR review)
- [x] mafft (Seems to work fine- @chriswyatt1, confirmed by @muffato )
- [x] maltextract @chriswyatt1 (need PR review)
- [x] maxquant/lfq (not failing)
- [x] metaphlan3/mergemetaphlantables #3976 
- [x] motus/downloaddb #2631
- [x] motus/merge #2631 (Conda failure dismissed https://github.com/nf-core/modules/pull/2631#issuecomment-1345626513 )
- [x] motus/profile #2631 (Conda failure dismissed https://github.com/nf-core/modules/pull/2631#issuecomment-1345626513 )
- [x] nanoplot #2610
- [x] ncbigenomedownload @chriswyatt1 
- [x] pairtools/dedup #3247
- [x] pairtools/flip #3247 
- [x] pairtools/restrict #3247 
- [x] picard/sortvcf @pawelciurkaardigen 
- [x] pirate @rpetit3
- [x] plasmidid #3746
- [ ] plink2/extract - tracked in #3965 
- [x] rtgtools/vcfeval (not failing, cf #2734)
- [x] shovill @rpetit3
- [x] sistr @rpetit3
- [x] stranger @pawelciurkaardigen (singularity and docker pass, for conda there is a problem creating conda environment - PackagesNotFound) 
- [x] strelka/somatic (not failing)
- [x] tbprofiler/profile @rpetit3
- [x] ultra/pipeline #2479 
- [ ] unicycler - tracked in #3982 
- [x] yara/mapper (#2164)
- [x] ariba/run @rpetit3
- [x] busco @rpetit3
- [x] dragonflye @rpetit3
- [x] mobsuite/recon @rpetit3
- [x] panaroo/run @rpetit3
- [x] prokka @rpetit3
- [x] roary @rpetit3
- [x] snippy/core @rpetit3
- [x] snippy/run @rpetit3

**Comment by amasplund** [comment]
We have looked at bcftools/sort and it looks like the md5sum of the zipped output vcf-file is different if you use a conda or docker profile. If you unzip the files the md5sum is the same. The docker one is the same as what is stated in the tests/modules/nf-core/bcftools/sort/test.yml file. Seems like a problem that should probably have appeared before!? Is there a standard solution? Remove the md5sum check of the zipped file??

**Comment by pawelciurkaardigen** [comment]
@amasplund I had similar problem for gatk4/applybqsr and validating cram file's md5sum but for me it was depending if I run the test locally or in CI. My solution was, as you suggested, relaxing the assertion by removing md5sum which is validating only file's presence. I don't know however if there is a better way that maintain file's content verification.

**Comment by muffato** [comment]
I found quite a few that have been fixed in the past few weeks

**Comment by muffato** [comment]
Re. unicycler: the Conda version comes with spades v3.14.1 whereas the Singularity container has spades v3.13.0.
v3.14.1 fails with the error
```
Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/opt/conda/conda-bld/spades_1616785826085/work/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.
Verification of expression 'cov_.size() > 10' failed in function 'void coverage_model::KMerCoverageModel::Fit()'. In file '/opt/conda/conda-bld/spades_1616785826085/work/src/common/modules/coverage_model/kmer_coverage_model.cpp' on line 187. Message 'Invalid kmer coverage histogram, make sure that the coverage is indeed uniform'.
spades-core: /opt/conda/conda-bld/spades_1616785826085/work/src/common/modules/coverage_model/kmer_coverage_model.cpp:187: void coverage_model::KMerCoverageModel::Fit(): Assertion `cov_.size() > 10' failed.
```

**Comment by muffato** [comment]
Finally getting to the bottom of this !

1. I've forked two issues for broken tests: #3965 and #3982 
2. The `deeparg/downloaddata` test is failing because the underlying deeparg version, 1.0.2, needs data from a website that is gone. There is no command-line parameter to change the URL and there is no other version. I've sent an email to the author about it. 
3. The `deeparg/predict` test fails because it first runs `deeparg/downloaddata`

My goal is to close this ticket once I've got an answer from the deeparg author.
